vpr/src/pack/cluster_legalizer.h

```diff
@@ -377,6 +383,18 @@ class ClusterLegalizer {
      */
     void reset();
 
+    /*
+     * @brief Checks if the given molecule is compatible with the given cluster.
+     *
+     * A molecule is compatible with a cluster if there exists "free primitives"
+     * (primitives that are not currently occupied by other atoms) for each atom
+     * in the molecule.
+     *
+     * This is a quick check to see if a molecule can go in the given cluster.
```

Comment:
I'd expand this a bit ... I'd add:

"This is a necessary but not sufficient test for a molecule to be able to go in a cluster. By calling it you can save runtime for impossible cases vs. calling the full checks."

Also change to "if there exists a free primitive (a primitive that is not currently occupied by other atoms) of the correct type to accomodate each type of atom in the molecule"